python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -343,6 +355,7 @@ def _retrieve_rows(self,
                        metric: Optional[str] = None,
                        mode: Optional[str] = None) -> Dict[str, Any]:
         assert mode is None or mode in ["max", "min"]
+        assert not mode or metric
```

Comment:
Is this supposed to be `!mode or metric` or `!(mode or metric)`? Could we use parenthesis here to make it clearer?

Reply:
It is supposed to be `!mode or metric` i.e. if mode is set, metric has to be set as well